cluster-autoscaler/core/scaledown/planner/controller.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package planner
+
+import (
+	"errors"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/autoscaler/cluster-autoscaler/utils/kubernetes"
+)
+
+var (
+	errNotFound = errors.New("owner not found")
+)
+
+type replicasInfo struct {
+	targetReplicas, currentReplicas int32
+}
+
+// ControllerReplicasCalculator calculates a number of target and expected replicas for a given controller.
+type ControllerReplicasCalculator interface {
```

Comment:
I think in general interfaces belong in the place where they are being used, not next to one of implementations. Also, why public? `newControllerCalculator` is private, so it is hard to use this interface outside of `planner` module anyway.

Reply:
moved to planner.go